osgp/platform/osgp-adapter-ws-core/src/main/java/org/opensmartgridplatform/adapter/ws/core/application/services/FirmwareManagementService.java

```diff
@@ -41,11 +45,12 @@
 import org.opensmartgridplatform.domain.core.exceptions.ExistingEntityException;
 import org.opensmartgridplatform.domain.core.exceptions.UnknownEntityException;
 import org.opensmartgridplatform.domain.core.repositories.FirmwareModuleRepository;
-import org.opensmartgridplatform.shared.validation.Identification;
 import org.opensmartgridplatform.domain.core.valueobjects.DeviceFunction;
 import org.opensmartgridplatform.domain.core.valueobjects.FirmwareModuleData;
 import org.opensmartgridplatform.domain.core.valueobjects.FirmwareUpdateMessageDataContainer;
 import org.opensmartgridplatform.domain.core.valueobjects.PlatformFunction;
+import org.opensmartgridplatform.dto.valueobjects.FirmwareModuleType;
```

Comment:
Originally DTOs were only used in between the domain and protocol adapters, but not in the WS adapter?
Aren't there any similar objects in domain.core available?

Reply:
There are classes FirmwareVersion and FirmwareModuleType in org.opensmartgridplatform.domain.core.valueobjects.smartmetering. Since this is specific smartmetering, I don't think we should use them.

Reply:
We could consider moving them to org.opensmartgridplatform.domain.core.valueobjects

Reply:
Moved FirmwareVersion and FirmwareModuleType  to org.opensmartgridplatform.domain.core.valueobjects; added orika mapping for these types and their dto counterparts to the core domain and make use of the mapping, so a response message on getFirmwareVersion contains a list of FirmwareVersion instead of FirmwareVersionDto. By doing this there is no need for a FirmwareVersionDto when handling the response message in the WS adapter.